Docs: Made clear-cut usage of laser.propagate (#157)
LASY-org/lasy: docs/source/user_guide/index.rst

#############

We will try a simple example to get familiar with the code structure and to verify the installation was successful.

> Lets generate a Gaussian pulse at focus, emit it behind the focal plan by one Rayleigh length and then output it to a file.

Reviewer: Numerically, an array is allocated and initialised with the laser profile at focus. Then, using a spectral method, the laser is actually propagated backward, in 1 large time step `\Delta t = - z_R/c`. Therefore, it is my impression that the word propagate here is justified. Is the wording below better?
```suggestion
Let's generate a Gaussian pulse at focus, propagate it backwards by one Rayleigh length (the pulse is then located ahead of the focal plane) and then output it to a file.
```

Reviewer: For clarification, one could add a sentence here, explaining that the output file can then be used as an input for a simulation tool, what do you think?

Author: It is already mentionned that: "This file may now be viewed, copied, shared or used as an input to a variety of other simulation tools." (at the end of the page)

> Let's generate a Gaussian pulse at focus, propagate it backwards by one Rayleigh length (the pulse is then located ahead of the focal plane) and then output it to a file.

Reviewer: ```suggestion
Let's generate a Gaussian pulse at focus, propagate it backwards by one Rayleigh length (the pulse is then located upstream of the focal plane) and then output it to a file.
```

   laser = Laser(dimensions,lo,hi,num_points,laser_profile)

..  code-block:: python

>    :caption: By default, the laser antenna will emit on the focal plan. (Optional) Make the laser antenna emit outside the focal plan by one Rayleigh length.

Reviewer: LASY does not really have an antenna (defined as a plane onto which we generate currents so as to create the desired electromagnetic wave). IMO, we should therefore avoid this word here, so I would have a similar comment to the point above. But I agree with you, it would be nice to make this a bit more clear: I mentioned this above.

Author: "By default, the values of the laser envelope are injected on the focal plan" does that sound better?

   z_R            = 3.14159*spot_size**2/wavelength    # The Rayleigh length.

>    laser.propagate(-z_R)                               # The laser antenna emits behind the focal plane

Reviewer: Likewise, this step actually does a numerical propagation by 1 large time step, and this is not use an antenna per se, so I think this wording could be confusing.

   laser = Laser(dimensions,lo,hi,num_points,laser_profile)

..  code-block:: python
   :caption: By default, the values of the laser envelope are injected on the focal plan. One can propagate it backwards by one Rayleigh length (optional).

   z_R            = 3.14159*spot_size**2/wavelength    # The Rayleigh length

>    laser.propagate(-z_R)                               # Propagate the pulse ahead of the focal plane

Reviewer: ```suggestion
   laser.propagate(-z_R)                               # Propagate the pulse upstream of the focal plane
```
